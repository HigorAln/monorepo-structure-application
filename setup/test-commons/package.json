{
  "name": "@alura/test-commons",
  "version": "0.0.0",
  "main": "index.js",
  "license": "MIT",
  "files": [
    "base.ts"
  ],
  "devDependencies": {
    "jest": "29.3.1",
    "ts-node": "10.9.1",
    "babel-jest": "29.3.1",
    "@babel/core": "7.20.2",
    "@babel/preset-env": "7.20.2",
    "@babel/preset-react": "7.18.6",
    "@babel/preset-typescript": "7.18.6",
    "@testing-library/react": "^13.4.0",
    "@testing-library/jest-dom": "^5.16.5"
  },
  "scripts": {
    "test:": "echo 'Not necessary'",
    "lint": "echo 'Not necessary'"
  }
}
