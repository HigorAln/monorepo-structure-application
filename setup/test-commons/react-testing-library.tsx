import { render, screen, renderHook } from '@testing-library/react'

export {
  render,
  screen,
  renderHook
}
