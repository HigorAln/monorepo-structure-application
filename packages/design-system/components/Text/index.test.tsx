import { Text } from "./index";
import { render, screen } from "@alura/test-commons/react-testing-library";

describe("<Text />", () => {
  it("render h1 tag", () => {
    const { container } = render(<Text tag="h1">Simple Text</Text>);

    expect(container).toMatchSnapshot();
  });

  it("render h2 tag", () => {
    render(<Text tag="h1">Simple Text</Text>);

    expect(screen.getByRole("heading")).toBeInTheDocument();
  });
});
