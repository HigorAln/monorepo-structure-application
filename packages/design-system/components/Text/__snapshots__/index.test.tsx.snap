// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> render h1 tag 1`] = `
<div>
  <h1
    class="sc-bcXHqe fjgEof"
  >
    Simple Text
  </h1>
</div>
`;
